Find a functionally equivalent code:android.app.Instrumentation.finish:COMMENT
Method Modifier: public      
Comment:/**
 * Terminate instrumentation of the application.  This will cause the
 * application process to exit, removing this instrumentation from the next
 * time the application is started.
 *
 * @param resultCode Overall success/failure of instrumentation.
 * @param results Any results to send back to the code that started the
 * instrumentation.
 */

@@ -3,8 +3,11 @@
         endPerformanceSnapshot();
     }
     if (mPerfMetrics != null) {
+        if (results == null) {
+            results = new Bundle();
+        }
         results.putAll(mPerfMetrics);
     }
     if (mUiAutomation != null) {

